appserver/microprofile/health/src/main/java/org/glassfish/microprofile/health/HealthServlet.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2024 Contributors to Eclipse Foundation.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package org.glassfish.microprofile.health;
+
+import jakarta.enterprise.inject.Instance;
+import jakarta.enterprise.inject.spi.CDI;
+import jakarta.json.bind.Jsonb;
+import jakarta.json.bind.JsonbBuilder;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.Liveness;
+import org.eclipse.microprofile.health.Readiness;
+import org.eclipse.microprofile.health.Startup;
+
+public class HealthServlet extends HttpServlet {
+
+    private static final String MP_DEFAULT_STARTUP_EMPTY_RESPONSE = "mp.health.default.startup.empty.response";
+    private static final String MP_DEFAULT_READINESS_EMPTY_RESPONSE = "mp.health.default.readiness.empty.response";
+    private static final Logger LOGGER = Logger.getLogger(HealthServlet.class.getName());
+
+    private static HealthCheckResponse callHealthCheck(HealthCheck healthCheck) {
+        try {
+            return healthCheck.call();
+        } catch (RuntimeException e) {
+            LOGGER.log(Level.SEVERE, "Health check failed", e);
+            return buildHealthCheckResponse(healthCheck.getClass().getName(), e);
+        }
+    }
+
+    private static HealthCheckResponse buildHealthCheckResponse(String name, Exception e) {
+        return HealthCheckResponse.builder()
+                .down()
+                .name(name)
+                .withData("rootCause", e.getMessage())
+                .build();
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
```

Comment:
Can we disable the health check endpoint with a system property? For example, return 404 status if disabled? 
It might be a security issue if the health endpoint is publicly available and wouldn't be possible to disable it.

Later we can add security to allow access to the endpoint only to authorized callers, but now it's enough if it's possible to disable the endpoint.